.github/actions/setup-linux/action.yml

```diff
@@ -30,16 +30,21 @@ runs:
         fi
 
     - name: Log in to ECR
-      shell: bash
+      uses: nick-fields/retry@3e91a01664abd3c5cd539100d10d33b9c5b68482
       env:
         AWS_RETRY_MODE: standard
         AWS_MAX_ATTEMPTS: "5"
         AWS_DEFAULT_REGION: us-east-1
-      run: |
-        AWS_ACCOUNT_ID=$(aws sts get-caller-identity|grep Account|cut -f4 -d\")
-        retry () { "$@"  || (sleep 1 && "$@") || (sleep 2 && "$@") }
-        retry aws ecr get-login-password --region "$AWS_DEFAULT_REGION" | docker login --username AWS \
-            --password-stdin "$AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com"
+      with:
+        shell: bash
+        timeout_minutes: 5
+        max_attempts: 3
+        retry_wait_seconds: 30
+        command: |
+          retry () { "$@"  || (sleep 1 && "$@") || (sleep 2 && "$@") }
```

Comment:
It looks like this has a custom retry function (but it didn't work, so retry might or might not help).  At least the GH action has a longer retry interval.  I think you need to clean up the retry function though.  Otherwise, it would be confusing

Reply:
I don't believe the problem was the retry not working.  There are two aws commands in this step, and the original version only had the retry invoked on one of them.